src/test/java/org/jabref/gui/externalfiles/FileFilterUtilsTest.java

```diff
@@ -1,16 +1,121 @@
+        try{
+            String dirPath = "src/test/resources/unlinkedFiles";
+            Path path = Paths.get(dirPath);
+            Files.createDirectory(path);
+
+            Path firstPath = Paths.get(dirPath.concat("/firstFile.pdf"));
+            Path secondPath = Paths.get(dirPath.concat("/secondFile.pdf"));
+            Path thirdPath = Paths.get(dirPath.concat("/thirdFile.pdf"));
+            Path fourthPath = Paths.get(dirPath.concat("/fourthFile.pdf"));
+
+            File firstFile = new File(firstPath);
+            File secondFile = new File(secondPath);
+            File thirdFile = new File(thirdPath);
+            File fourthFile = new File(fourthPath);
+
+            firstFile.createNewFile();
+            secondFile.createNewFile();
+            thirdFile.createNewFile();
+            fourthFile.createNewFile();
+
+            SimpleDateFormat dateFormat = new SimpleDateFormat("mm/dd/yyyy");
+
+            Date firstDate = dateFormat.parse("10/1/2021");
+            Date secondDate = dateFormat.parse("5/1/2021");
+            Date thirdDate = dateFormat.parse("1/1/2021");
+            Date fourthDate = dateFormat.parse("2/1/2021");
+
+            firstFile.setLastModified(firstDate.getTime());
+            secondFile.setLastModified(secondDate.getTime());
+            thirdFile.setLastModified(thirdDate.getTime());
+            fourthFile.setLastModified(fourthDate.getTime());
+
+            test.files.add(firstPath);
+            test.files.add(secondPath);
+            test.files.add(thirdPath);
+            test.files.add(fourthPath);
+
+            test.expectedSortByDateAscending.add(firstPath.toString());
+            test.expectedSortByDateAscending.add(secondPath.toString());
+            test.expectedSortByDateAscending.add(thirdPath.toString());
+            test.expectedSortByDateAscending.add(fourthPath.toString());
+
+            test.expectedSortByDateDescending.add(Paths.concat("/firstFile.pdf")));
+            test.expectedSortByDateDescending.add(Paths.concat("/secondFile.pdf")));
+            test.expectedSortByDateDescending.add(Paths.concat("/thirdFile.pdf")));
+            test.expectedSortByDateDescending.add(Paths.concat("/fourthFile.pdf")));
+        } catch (Exception e) {}
+    }
+
+    @AfterAll
+    private static void cleanUp() {
+        String dirPath = "src/test/resources/unlinkedFiles";
+
+        File firstFile = new File(dirPath.concat("/firstFile.pdf"));
+        File secondFile = new File(dirPath.concat("/secondFile.pdf"));
+        File thirdFile = new File(dirPath.concat("/thirdFile.pdf"));
+        File fourthFile = new File(dirPath.concat("/fourthFile.pdf"));
+        File directory = new File(dirPath);
+        
+        firstFile.delete();
+        secondFile.delete();
+        thirdFile.delete();
+        fourthFile.delete();
+        directory.delete();
+    }
+
+    @Test
+    public void testSortByDateAscending() {
+        ArrayList<String> actuals = new ArrayList<String>();
```

Comment:
Code against Interfaces,  not implementations  https://medium.com/javarevisited/oop-good-practices-coding-to-the-interface-baea84fd60d3 
Therefore, the Left side should always be the most "abstract" type (or an interface)

Reply:
Hello @Siedlerchr thanks for the quick response, those tests are under construction and we mistakenly commited them ahead of time. We plan to commit them when they are ready, but thanks for the help we will include your suggestions.